lib/solscan_api_ruby/methods/public.rb

```diff
@@ -0,0 +1,304 @@
+module SolscanApiRuby
+  module Methods
+    class Public < Base
+      # @see https://public-api.solscan.io/docs/#/Block/get_block_last
+      #
+      # Get last [limit] blocks
+      #
+      # @param limit [Integer]
+      #
+      # @return [Response, ApiError] Response when success, ApiError on failure.
+      def block_last(limit: nil)
+        params = {}
+
+        params[:limit] = limit if limit
+
+        send_request(__method__, params: params)
+      end
+
+      # @see https://public-api.solscan.io/docs/#/Block/get_block_transactions
+      #
+      # Get block transactions. MaxLimit 50 records per request
+      #
+      # @param block [Integer]
+      # @param offset [Integer]
+      # @param limit [Integer]
+      #
+      # @return [Response, ApiError] Response when success, ApiError on failure.
+      def block_transactions(block, offset: nil, limit: nil)
+        params = {}
+
+        params[:block] = block
+        params[:offset] = offset if offset
+        params[:limit] = limit if limit
```

Comment:
if max limit is 50 then perhaps this should be handled somehow? We do not have to send request with exceeded limit and wait for error from solscan.

Reply:
Solscan does not return error if you pass limit higher than 50, it returns error when payload is too big, but it returns sometimes ie. 100 records if you set limit to this (checked here: https://public-api.solscan.io/docs/#/Account/get_account_transactions).

 So I am going to leave it as is.